Index: mt7620-p4rev-120998/src/ap/ap_assoc.c
===================================================================
--- mt7620-p4rev-120998.orig/src/ap/ap_assoc.c
+++ mt7620-p4rev-120998/src/ap/ap_assoc.c
@@ -321,7 +321,8 @@ VOID ap_cmm_peer_assoc_req_action(
 
 	DBGPRINT(RT_DEBUG_TRACE, ("%s - MBSS(%d), receive %s request from %02x:%02x:%02x:%02x:%02x:%02x\n",
 							  sAssoc, pEntry->apidx, sAssoc, PRINT_MAC(Addr2)));
-    
+   	
+	printk("%s: %s from %02x:%02x:%02x:%02x:%02x:%02x\n", pAd->net_dev->name, sAssoc, PRINT_MAC(Addr2));	
     /* supported rates array may not be sorted. sort it and find the maximum rate */
     for (i=0; i<SupportedRatesLen; i++)
     {
Index: mt7620-p4rev-120998/src/ap/ap_auth.c
===================================================================
--- mt7620-p4rev-120998.orig/src/ap/ap_auth.c
+++ mt7620-p4rev-120998/src/ap/ap_auth.c
@@ -211,6 +211,7 @@ static VOID APPeerDeauthReqAction(
 				 "%02x:%02x:%02x:%02x:%02x:%02x, reason=%d\n", SeqNum,
 				Addr2[0], Addr2[1], Addr2[2], Addr2[3], Addr2[4], Addr2[5], Reason));
 
+		printk("%s: DE-AUTH from %02x:%02x:%02x:%02x:%02x:%02x\n", pAd->net_dev->name, PRINT_MAC(Addr2));
 #ifdef MAC_REPEATER_SUPPORT
 		if (pAd->ApCfg.bMACRepeaterEn == TRUE)
 		{
