Index: mt76x2e-p4rev-113843/src/common/rtmp_init.c
===================================================================
--- mt76x2e-p4rev-113843.orig/src/common/rtmp_init.c
+++ mt76x2e-p4rev-113843/src/common/rtmp_init.c
@@ -3479,7 +3479,7 @@ VOID RTMP_TimerListAdd(RTMP_ADAPTER *pAd
 	{
 		pObj->pRscObj = pRsc;
 		insertTailList(pRscList, (LIST_ENTRY *)pObj);
-		DBGPRINT(RT_DEBUG_WARN, ("%s: add timer obj %lx!\n", __FUNCTION__, (ULONG)pRsc));
+		//DBGPRINT(RT_DEBUG_WARN, ("%s: add timer obj %lx!\n", __FUNCTION__, (ULONG)pRsc));
 	}
 }
 
@@ -3510,7 +3510,7 @@ VOID RTMP_TimerListRelease(RTMP_ADAPTER
 		delEntryList(pRscList, pListEntry);
 
 		/* free a timer record entry */
-		DBGPRINT(RT_DEBUG_ERROR, ("%s: release timer obj %lx!\n", __FUNCTION__, (ULONG)pRsc));
+		//DBGPRINT(RT_DEBUG_ERROR, ("%s: release timer obj %lx!\n", __FUNCTION__, (ULONG)pRsc));
 		os_free_mem(NULL, pObj);
 	}
 }
